driftbase/api/flexmatch.py

```diff
@@ -0,0 +1,117 @@
+    matchmaker = fields.String(required=True, description="Which matchmaker (configuration name) to issue the ticket for. ")
+
+class FlexMatchPlayerAPIPutArgs(Schema):
+    match_id = fields.String(required=True, description="The id of the match being accepted/rejected")
+    acceptance = fields.Boolean(required=True, description="True if match_id is accepted, False otherwise")
+
+
+@bp.route("/")
+class FlexMatchPlayerAPI(MethodView):
+
+    @bp.arguments(FlexMatchPlayerAPIPatchArgs)
+    def patch(self, args):
+        """
+        Add a freshly measured latency value to the player tally.
+        Returns a region->avg_latency mapping.
+        """
+        player_id = current_user["player_id"]
+        latency = args.get("latency_ms")
+        region = args.get("region")
+        if None in (latency, region) or region not in flexmatch.VALID_REGIONS or not isinstance(latency, (int, float)):
+            abort(http_client.BAD_REQUEST, message="Invalid or missing arguments")
+        flexmatch.update_player_latency(player_id, region, latency)
+        return flexmatch.get_player_latency_averages(player_id), http_client.OK
+
+    @bp.arguments(FlexMatchPlayerAPIPostArgs)
+    def post(self, args):
+        """
+        Insert a matchmaking ticket for the requesting player or his party.
+        Returns a ticket.
+        """
+        try:
+            ticket = flexmatch.upsert_flexmatch_ticket(current_user["player_id"], args.get("matchmaker"))
+            return ticket, http_client.OK
+        except flexmatch.GameliftClientException as e:
+            log.error(f"Inserting/updating matchmaking ticket for player {current_user['player_id']} failed: Gamelift response:\n{e.debugs}")
+            return {"error": e.msg}, http_client.INTERNAL_SERVER_ERROR
+
+    @bp.arguments(FlexMatchPlayerAPIPutArgs)
+    def put(self, args):
+        """
+        Accept or decline a match
+        """
+        player_id = current_user["player_id"]
+        match_id = args.get("match_id")
+        acceptance = args.get("acceptance")
+        try:
+            flexmatch.update_player_acceptance(player_id, match_id, acceptance)
+            return {}, http_client.OK
+        except flexmatch.GameliftClientException as e:
+            log.error(f"Updating the acceptance status for {match_id} on behalf of player {player_id} failed: Gamelift response:\n{e.debugs}")
+            return {"error": e.msg}, http_client.INTERNAL_SERVER_ERROR
+
+    def get(self):
+        """
+        Retrieve the active matchmaking ticket for the requesting player or his party, or empty dict if no such thing is found.
+        """
+        ticket = flexmatch.get_player_ticket(current_user["player_id"])
+        return ticket or {}, http_client.OK
+
+    def delete(self):
+        """
+        Delete the currently active matchmaking ticket for the requesting player or his party.
+        """
+        try:
+            deleted_ticket = flexmatch.cancel_player_ticket(current_user["player_id"])
```

Comment:
Don't think it's worthwhile; firstly, the client shouldn't be able to press the cancel button once the ticket enters a non-cancellable state and secondly, the client shouldn't be acting on the response at all.  He should get all actionable state changes via events.   Extra logging might be a thing....

Reply:
What if you're in a party and two players cancel?

Reply:
It would've resulted in one of them getting a 404, which is the correct behavior imo.  But I put in a http.OK with the status if someone tries to delete a ticket in a 'matched' state.